pkg/imagehandler/imagehandler.go

```diff
@@ -82,7 +94,7 @@ func (f *imageFileSystem) getNameForKey(key string) (name string, err error) {
 func (f *imageFileSystem) ServeImage(key string, ignitionContent []byte, initramfs, static bool) (string, error) {
 	size, err := f.getBaseImage(initramfs).Size()
 	if err != nil {
-		return "", err
+		return "", InvalidBaseImageError{cause: err}
```

Comment:
For completeness, we could add a check to make sure that in addition to there being no errors, the size is also not zero.